Mixpanel/Mixpanel.m

```diff
@@ -2072,22 +1951,24 @@ - (void)addPeopleRecordToQueueWithAction:(NSString *)action andProperties:(NSDic
                 r[action] = [NSDictionary dictionaryWithDictionary:p];
             }
             
+            NSMutableArray *peopleQueue = strongMixpanel.peopleQueue;
             if (self.distinctId) {
                 r[@"$distinct_id"] = self.distinctId;
                 MixpanelDebug(@"%@ queueing people record: %@", self.mixpanel, r);
-                [strongMixpanel.peopleQueue addObject:r];
-                if (strongMixpanel.peopleQueue.count > 500) {
-                    [strongMixpanel.peopleQueue removeObjectAtIndex:0];
+                [peopleQueue addObject:r];
+                if (peopleQueue.count > 500) {
+                    [peopleQueue removeObjectAtIndex:0];
                 }
+                MPPersistence *persistence = strongMixpanel.persistence;
+                [persistence archivePeopleQueue:peopleQueue];
```

Comment:
why do we move this logic inside this `if`? I think we want to archive people if `people.identify` isn't called yet.